README.md

```diff
@@ -16,25 +16,133 @@ specified bucket in GCS.
 ### File name format
 
 The connector uses the following format for output files (blobs):
-`<prefix><topic>-<partition>-<start_offset>[.gz]`, where:
-- `<prefix>` is the optional prefix that can be used, for example, for
-  subdirectories in the bucket;
-- `<topic>` is the Kafka topic name;
-- `<partition>` is the topic's partition number;
-- `<start_offset>` is the Kafka offset of the first record in the file;
-- `[.gz]` suffix is added when compression is enabled.
-
-The part after the prefix can be controlled by `file.name.template`. It
-supports placeholders with variable names: `{{ variable_name }}`. Only
-certain sets of variables are valid, currently:
-- `topic`, `partition`, `start_offset`.
-
-If not specified, the default value is
+`<prefix><filename>`.
+
+`<prefix>` is the optional prefix that can be used, for example, for
+subdirectories in the bucket.
+
+`<filename>` is the file name. The connector has the configurable
+template for file names. It supports placeholders with variable names:
+`{{ variable_name }}`. Currently supported variables are:
+- `topic` - the Kafka topic;
+- `partition` - the Kafka partition;
+- `start_offset` - the Kafka offset of the first record in the file;
+- `key` - the Kafka key.
+
+Only the certain combinations of variables are allowed in the file name
+template (however, variables in a template can be in any order). Each
+combination determines the mode of record grouping the connector will
+use. Currently supported combinations of variables and the corresponding
+record grouping modes are:
+- `topic`, `partition`, `start_offset` - grouping by the topic and
+  partition;
+- `key` - grouping by the key.
+
+If the file name template is not specified, the default value is
 `{{topic}}-{{partition}}-{{start_offset}}` (+ `.gz` when compression is
 enabled).
 
-Even when `file.name.template` is explicitly set, the prefix is still
-added.
+### Record grouping
+
+Incoming records are being grouped until flushed.
+
+#### Grouping by the topic and partition
+
+In this mode, the connector groups records by the topic and partition.
+When a file is written, a offset of the first record in it is added to
+its name.
+
+For example, let's say the template is
+`t{{topic}}-p{{partition}}-o{{start_offset}}`. If the connector receives
+records like
+```
+topic:1 partition:0 offset:1
```

Comment:
Non-numeric sample values for topics might be better since topic names can and usually are something else than numbers.